libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -47,9 +46,17 @@ void AP_Frsky_Telem::init(const AP_SerialManager &serial_manager, const char *fi
         queue_message(MAV_SEVERITY_INFO, firmware_str);
         // save main parameters locally
         _params.mav_type = mav_type; // frame type (see MAV_TYPE in Mavlink definition file common.h)
-        _params.fs_batt_voltage = fs_batt_voltage; // failsafe battery voltage in volts
-        _params.fs_batt_mah = fs_batt_mah; // failsafe reserve capacity in mAh
-        _ap.value = ap_value; // ap bit-field
+        if (fs_batt_voltage != nullptr) {
+            _params.fs_batt_voltage = fs_batt_voltage; // failsafe battery voltage in volts
+        }
+        if (fs_batt_mah != nullptr) {
+            _params.fs_batt_mah = fs_batt_mah; // failsafe reserve capacity in mAh
+        }
+        if (ap_value != nullptr) {
+            _ap.value = ap_value; // ap bit-field
+        } else {
+            *_ap.value = 0;
```

Comment:
This is wrong, you are assigning a value to an unknown location (in the case of ArduPilot the location is known to be 0, but you should never do this).

Also what is the need for all the ifs? If it receives nullptr there's no harm in assigning nullptr...